var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,531 @@
+                '-DKLU_ENABLE=ON',
+                '-DKLU_INCLUDE_DIR={0}'.format(
+                    spec['suite-sparse'].prefix.include),
+                '-DKLU_LIBRARY_DIR={0}'.format(
+                    spec['suite-sparse'].prefix.lib),
             ])
 
+        # Building with SuperLU_MT
+        if '+superlu-mt' in spec:
+            args.extend([
+                '-DSUPERLUMT_ENABLE=ON',
+                '-DSUPERLUMT_INCLUDE_DIR={0}'.format(
+                    spec['superlu-mt'].prefix.include),
+                '-DSUPERLUMT_LIBRARY_DIR={0}'.format(
+                    spec['superlu-mt'].prefix.lib)
+            ])
+            if '+openmp' in spec:
+                args.append('-DSUPERLUMT_THREAD_TYPE=OpenMP')
+            elif '+pthread' in spec:
+                args.append('-DSUPERLUMT_THREAD_TYPE=Pthread')
+            else:
+                msg = 'You must choose either +openmp or +pthread when '
+                msg += 'building with SuperLU_MT'
+                raise RuntimeError(msg)
+
         # Building with Hypre
-        if '+hypre' in spec and spec.satisfies('@2.7:'):
-            cmake_args.extend([
+        if '+hypre' in spec:
+            args.extend([
+                '-DHYPRE_ENABLE=ON',
                 '-DHYPRE_INCLUDE_DIR={0}'.format(
                     spec['hypre'].prefix.include),
                 '-DHYPRE_LIBRARY_DIR={0}'.format(
                     spec['hypre'].prefix.lib)
             ])
 
-        # Building with LAPACK and BLAS
+        # Building with PETSc
+        if '+petsc' in spec:
+            args.extend([
+                '-DPETSC_ENABLE=ON',
+                '-DPETSC_INCLUDE_DIR={0}'.format(
+                    spec['petsc'].prefix.include),
+                '-DPETSC_LIBRARY_DIR={0}'.format(
+                    spec['petsc'].prefix.lib)
+            ])
+
+        # Examples
+        if not spec.satisfies('@:2.7.0'):
+            args.extend([
+                '-DEXAMPLES_ENABLE_C=%s'    % on_off('+examples-c'),
+                '-DEXAMPLES_ENABLE_CXX=%s'  % on_off('+examples-cxx'),
+                '-DEXAMPLES_ENABLE_F77=%s'  % on_off('+examples-f77'),
+                '-DEXAMPLES_ENABLE_F90=%s'  % on_off('+examples-f90'),
+                '-DEXAMPLES_ENABLE_CUDA=%s' % on_off('+examples-cuda'),
+                '-DEXAMPLES_ENABLE_RAJA=%s' % on_off('+examples-raja')
+            ])
+        else:
+            args.extend([
+                '-DEXAMPLES_ENABLE=%s' % on_off('+examples-c'),
+                '-DCXX_ENABLE=%s'      % on_off('+examples-cxx'),
+                '-DF90_ENABLE=%s'      % on_off('+examples-f90')
+            ])
+
+        args.extend([
+            '-DEXAMPLES_INSTALL=%s' % on_off('+examples-install')
+        ])
+
+        return args
+
+    # ==========================================================================
+    # SUNDIALS xSDK Settings
+    # ==========================================================================
+
+    @when('xsdk-defaults=True')
```

Comment:
do you really need two different `cmake_args`? there is so much code duplication here!.. Can you not do some `if/else` within the same function based on `xsdk`?

Reply:
Looking over this again we don't really need the xsdk-defaults option in Spack since user's don't directly set the CMake variables. So I've removed this option and the related cmake_args function.